File: zetapushlibrary/src/main/java/com/zetapush/library/StorageCredentialsInterface.java
package com.zetapush.library;

import java.util.Map;

/**
 * Created by damien on 20/07/17.
 */

public interface StorageCredentialsInterface {

    /**
     * Save the credentials used for the simple authentication
     * @param login
     * @param password
     */
    void saveCredentials(String login, String password);

    /**
     * Get the credentials as a map with key 'login' and 'password'
     * @return
     */
    Map<String, String> getCredentials();

    /**
     * Clear the saved credentials
     */
    void clearCredentials();
}

File: android-library/src/main/java/com/zetapush/library/StorageCredentialsInterface.kt
package com.zetapush.library

data class Credentials(val login: String? = null, val password: String? = null)

interface StorageCredentialsInterface {

    /**
     * Get the credentials as a map with key 'login' and 'password'
     *
     * @return
     */
    val credentials: Credentials?

    /**
     * Save the credentials used for the simple authentication
     *
     * @param login
     * @param password
     */
    fun saveCredentials(login: String?, password: String?)

    /**
     * Clear the saved credentials
     */
    fun clearCredentials()
}
